File: front-end/src/app/core/enums/book-order-field.ts
/** Available field used for ordering books. */
export enum BookOrderField {
  Title = 'title',
  Price = 'price',
  PublicationDate = 'published',
}

export namespace BookOrderField {
  const TO_READABLE_MAP: Readonly<Record<BookOrderField, string>> = {
    [BookOrderField.Price]: 'Price',
    [BookOrderField.Title]: 'Title',
    [BookOrderField.PublicationDate]: 'Publication Date',
  };

  /**
   * Maps to enum value into a human-readable representation of it.
   * @param field Enum value to map into readable one.
   */
  export function toReadable(field: BookOrderField): string {
    return TO_READABLE_MAP[field];
  }
}

File: back-end/src/main/java/com/huyphan/models/enums/BookOrderField.java
package com.huyphan.models.enums;

import com.fasterxml.jackson.annotation.JsonValue;

/** Available field for ordering books. */
public enum BookOrderField {
	Title("title"), Price("price"), PublicationDate("published");

	private String value;

	private BookOrderField(String value) {
		this.value = value;
	}

	/** Return string value of this enum. */
	@JsonValue
	public String getValue() {
		return value;
	}


}
